Make the Mole, Root Sapling, and Corrupted Big Granito a StickyBadguy

swagtoy opened the pull request:
Fixes #3132

I did some basic testing, but I need someone to properly test the stickiness.
Thank you!

swagtoy committed:
Refactor StickyBadguy to support other sprites

Bad logic anyway, will be improved later.

swagtoy committed:
Make Corrupted granito, mole, and root sapling sticky

Rusty-Box committed:
Update corrupted_granito_big.cpp [ci skip]

Rusty-Box committed:
Update corrupted_granito_big.hpp

Rusty-Box committed:
Revert "stickyness" in corrupted_granito.cpp [ci skip]

Rusty-Box committed:
Revert "stickyness" in corrupted_granito.hpp

swagtoy commented:
Thx king. I think it's good to merge

weluvgoatz commented:
No?

swagtoy commented:
???

Rusty-Box commented:
@swagtoy B.tw. there were other instance of 'BadGuy' in 'corrupted_big_granito.cpp' in line 52 and 62. Wasn't sure if I also replace that with 'StickyBadguy' or not.

swagtoy commented:
Wait hold on this PR sucks a lot

swagtoy commented:
RE @Rusty-Box There is actually nothing wrong with calling Badguy::* functions, but for consistency I have made it go to the StickyBadguy in any situation where these functions may ever get overloaded

swagtoy committed:
Make stickyness sticky

I wasn't calling the actual sticky related functions like an idiot. I have discussed that this can be refactored one day anyway, so it's whatever...

swagtoy commented:
@Rusty-Box You can now test this again. I apparently didn't even call the sticky functions like an idiot. They are now getting called and I've also applied your suggestion.

swagtoy commented:
Just a note for myself: Need to fix Root Sapling and Corrupted Big Granito; these do not work in this PR.

MatusGuy committed:
fix constants

MatusGuy reviewed (changes requested):
Something's wrong. The stickiness setting is not being saved. This is the case with the Root Sapling. I'm not sure if this is the fault of this PR.

swagtoy commented:
I completely forgot about this PR lol
